embedding/csharp/Tizen.Flutter.Embedding.Tests/Channels/EventChannelTests.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright 2021 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using NSubstitute;
+using Xunit;
+
+namespace Tizen.Flutter.Embedding.Tests.Channels
+{
+    public class EventChannelTests
+    {
+        const string TEST_CHANNEL_NAME = "TEST/CHANNEL";
+
+        public class TheCtor
+        {
+            [Fact]
+            public void Ensures_Name_Is_Not_Null_Or_Empty()
+            {
+                var messenger = Substitute.For<IBinaryMessenger>();
+                Assert.Throws<ArgumentException>(() => new EventChannel("", StandardMethodCodec.Instance, messenger));
+                Assert.Throws<ArgumentException>(() => new EventChannel(null, StandardMethodCodec.Instance, messenger));
+            }
+
+            [Fact]
+            public void Ensures_Codec_Is_Not_Null()
+            {
+                var messenger = Substitute.For<IBinaryMessenger>();
+                Assert.Throws<ArgumentNullException>(() => new EventChannel(TEST_CHANNEL_NAME, null, messenger));
+            }
+
+            [Fact]
+            public void Ensures_Messenger_Is_Not_Null()
+            {
+                Assert.Throws<ArgumentNullException>(() => new EventChannel(TEST_CHANNEL_NAME, StandardMethodCodec.Instance, null));
+            }
+        }
+
+        public class TheSetStreamHandlerMethod
+        {
+            [Fact]
+            public void Registers_Stream_Handler()
+            {
+                var messenger = Substitute.For<IBinaryMessenger>();
+                var handler = Substitute.For<IEventStreamHandler>();
```

Comment:
How about using `new EventStreamHandler()` here?

Reply:
This test to ensure that the call is delivered properly to the `BinaryMessenger`. IMO, it is not necessary to create an `EventStreamHandler` instance directly. `Test double` is enough in this case.